force-app/main/default/classes/FSL/WorkOrderImportController.cls

```diff
@@ -0,0 +1,456 @@
+        }
+        List<Id> dbWorkOrderExcelAmountIds = createWorkOrders (
+            validatedAndPreparedWorkOrders,
+            workOrdersWithPropertyIdsAndContactIdsExcel
+            );
+        if (checkErrorsToFront(errorsToFrontList)) {
+            return JSON.serialize(errorsToFrontList);
+        }
+        if (dbWorkOrderExcelAmountIds == null) {
+            throw new AuraHandledException('Problem occured during creating Work Orders');
+        }
+        List<Id> dbWorkOrderLineItemIds = createWorkOrderLineItems (
+            workOrderData.WorkOrderLineItems,
+            dbWorkOrderExcelAmountIds
+            );
+        if (checkErrorsToFront(errorsToFrontList)) {
+            return JSON.serialize(errorsToFrontList);
+        }
+        if (dbWorkOrderLineItemIds == null) {
+            throw new AuraHandledException('Problem occured during creating Work Order Line Items');
+        }
+        List<Id> serviceAppointmentIds = createServiceAppointments (
+            workOrderData.ServiceAppointments,
+            dbWorkOrderLineItemIds
+            );
+        if (checkErrorsToFront(errorsToFrontList)) {
+            return JSON.serialize(errorsToFrontList);
+        }
+        if (serviceAppointmentIds == null) {
+            throw new AuraHandledException('Problem occured during creating Service Appointments');
+        }
+        return null;
+    }
+
+    private static List<WorkOrderImportModel.WorkOrderExcel> addWorkOrderPropertyIds (
+        List<WorkOrderImportModel.WorkOrderExcel> workOrderExcelList
+        ) {
+        Set <String> oldPropertyIdSet = new Set<String>();
+        for (WorkOrderImportModel.WorkOrderExcel w : workOrderExcelList) {
+            oldPropertyIdSet.add(w.PropertyId);
+        }
+        List<Property__c> properties = [SELECT Yardi_Property_ID__c
+                                        FROM Property__c
+                                        WHERE Yardi_Property_ID__c IN :oldPropertyIdSet];
+        Map<String, Id> propertyYardiIdMap = new Map<String, Id>();
+        for (Property__c p : properties) {
+            propertyYardiIdMap.put(p.Yardi_Property_ID__c, p.Id);
+        }
+        if (propertyYardiIdMap.keyset().size() < oldPropertyIdSet.size()) {
+            for (String pId : propertyYardiIdMap.keyset()) {
+                if (oldPropertyIdSet.contains(pId)) {
+                    oldPropertyIdSet.remove(pId);
+                }
+            }
+            for (WorkOrderImportModel.WorkOrderExcel w : workOrderExcelList) {
+                if (oldPropertyIdSet.contains(w.PropertyId)) {
+                    ErrorEntity e = new ErrorEntity('1',
+                                                    w.FakeId,
+                                                    new List<String>
+                                                    {'This PropertyId is missing in the DB'});
+                    errorsToFrontList.add(e);
+                }
+            }
```

Comment:
This should be extracted to external function.
Also you do not need to loop over all items - you just need to loop over oldPropertyIdSet - at this point this set contains only Ids of properties not found in the DB - if is empty, then no errors will be returned.

Reply:
But I need also FakeId field from the workOrderExcelList